canary/producer-c/canary/CanaryStreamUtils.cpp

```diff
@@ -297,6 +297,21 @@ STATUS publishErrorRate(STREAM_HANDLE streamHandle, PCanaryStreamCallbacks pCana
     return retStatus;
 }
 
+STATUS pushStartUpLatency(PCanaryStreamCallbacks pCanaryStreamCallbacks)
+{
+    Aws::CloudWatch::Model::MetricDatum startupLatencyDatum, aggstartupLatencyDatum;
+    startupLatencyDatum.SetMetricName("StartupLatency");
+    startupLatencyDatum.AddDimensions(pCanaryStreamCallbacks->dimensionPerStream);
+    pushMetric(pCanaryStreamCallbacks, startupLatencyDatum, Aws::CloudWatch::Model::StandardUnit::Milliseconds, pCanaryStreamCallbacks->startUpLatency);
+
+    aggstartupLatencyDatum.SetMetricName("StartupLatency");
+    aggstartupLatencyDatum.AddDimensions(pCanaryStreamCallbacks->aggregatedDimension);
+    pushMetric(pCanaryStreamCallbacks, aggstartupLatencyDatum, Aws::CloudWatch::Model::StandardUnit::Milliseconds, pCanaryStreamCallbacks->startUpLatency);
+
+CleanUp:
```

Comment:
Not sure if you need this label. It will generate warnings if not used. Either remove this or if the input is not sanitized add CHK(pCanaryStreamCallbacks != NULL, STATUS_BLAH);

Reply:
Makes sense. My bad. Will remove it.